sys/include/net/cord/lc.h

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (C) 2018 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_cord_lc CoRE RD Lookup Client
+ * @ingroup     net_cord
+ * @brief       Library for using RIOT as CoRE Resource Directory lookup client
+ *
+ * This module implements a CoRE Resource Directory lookup client library, that
+ * allows RIOT nodes to lookup resources, endpoints and groups with resource
+ * directories.
+ * It implements the standard lookup functionality as defined in
+ * draft-ietf-core-resource-directory-20.
+ * @see https://tools.ietf.org/html/draft-ietf-core-resource-directory-20
+ *
```

Comment:
I think it would be good to add a paragraph documenting the limitations of this implementation, e.g. that the user is not able to make more than a single request concurrently and that lookup function calls are fully synchronously.

By the way: there is actually no reason for this anymore. Back when I implemented `core_ep`, there were issues with `gcoap` that prevented me from using it asynchronously. But those are no longer existing, so we could actually rewrite the code to lift these limitations. But that is probably best done in some future work :-)

Reply:
Added doc documenting the limitation and also the different lookup modes available in 6f7d2e17a6ab3c38c3cfd2ffa15e5bb1e5dfb138.

> By the way: there is actually no reason for this anymore. Back when I implemented `core_ep`, there were issues with `gcoap` that prevented me from using it asynchronously. But those are no longer existing, so we could actually rewrite the code to lift these limitations. But that is probably best done in some future work :-)

:+1:

Reply:
> Back when I implemented `core_ep`, there were issues with `gcoap` that prevented me from using it asynchronously.

Perhaps an async approach would help with #12884.

Reply:
It seems to me that #12884 is unrelated to this and can be fixed on its own -> will comment there